fabtests/multinode/src/timing.c

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2022 Intel Corporation. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include <string.h>
+#include <timing.h>
+#include <core.h>
+#include <limits.h>
+
+extern struct pm_job_info pm_job;
+
+int multi_timer_create(struct multi_timer **timer, int rank)
+{
+	*timer = malloc(sizeof(*timer));
+
+	(*timer)->rank = rank;
+	(*timer)->start = -1;
+	(*timer)->end = -1;
+
+	return 0;
+}
+
+void multi_timer_start(struct multi_timer *timer)
+{
+	struct timespec spec;
+
+	if (timer && timer->start == -1) {
+		clock_gettime(CLOCK_REALTIME, &spec);
+		timer->start = spec.tv_nsec + spec.tv_sec * 1000000000;
+	}
+}
+
+void multi_timer_stop(struct multi_timer *timer)
+{
+	struct timespec spec;
+
+	if (timer && timer->start != -1 && timer->end == -1) {
+		clock_gettime(CLOCK_REALTIME, &spec);
+		timer->end = spec.tv_nsec + spec.tv_sec * 1000000000;
+	}
+}
```

Comment:
We have functions to get timestamps -- ft_gettime_ns() / us() / ms() -- which can be used instead, and call clock_gettime() correctly.  We shouldn't need either of the above start/stop functions.